bin/lint-php-changed.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+DEFAULT_BRANCH=$(git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@')
+INITIAL_COMMIT_FOR_BRANCH=$(git merge-base origin/$DEFAULT_BRANCH HEAD)
+CHANGED_PHP_FILES=$(git diff --name-only --diff-filter=d $INITIAL_COMMIT_FOR_BRANCH | grep '\.php$' | xargs)
+if [ -z "$CHANGED_PHP_FILES" ]; then
+	echo "No PHP files changed. Exiting."
+	exit 0
+fi
+vendor/bin/phpcs-changed --warning-severity=0 --extensions=php,html --git --git-staged $CHANGED_PHP_FILES
```

Comment:
I have a concern about the exact way this code is set up, mainly because the call to `phpcs-changed` is passing in `--git-staged`, but the logic above is finding all PHP files that have been changed on the current branch. 

I also notice that we're including `html` in the `--extensions` list despite limiting the linter to files that end in `.php` above.

If we're wanting to run this only for staged files, then it should be enough to have the changed files computed via the `--staged` argument to `git diff`. If we only want this for staged files, then it may be worth moving my suggestion directly into `.husky/pre-commit`. But if we want the code to work more broadly, we should consider pushing the changed file detection into separate scripts for basic approaches. e.g. for pre-push, we would probably want to consider all changes on the branch.
```suggestion
CHANGED_STAGED_PHP_FILES=$(git diff --staged --name-only --diff-filter=d | grep '\.php$' | xargs)
if [ -z "$CHANGED_STAGED_PHP_FILES" ]; then
	echo "No modified PHP files are staged. Exiting."
	exit 0
fi
vendor/bin/phpcs-changed --warning-severity=0 --extensions=php --git --git-staged $CHANGED_STAGED_PHP_FILES
```

Reply:
I observed that there are numerous other linters in the `package.json` file, which leads me to believe that keeping this code outside of `.husky/pre-commit` would be a better approach. If the team decides to do so, other linters (such as `npm run lint:js`) can be easily incorporated as additional lines within the `.husky/pre-commit` file.

Reply:
My bigger concern is that there is nothing in the naming to suggest that it will only operate on staged files. In general, linters will run on the files you give them, but this particular script will only lint staged files. Can you update the script name and alias to reflect that important distinction? Maybe `bin/lint-php-staged.sh` and `lint:php:staged`?